packages/gatsby/src/utils/websocket-manager.ts

```diff
@@ -178,6 +186,22 @@ export class WebsocketManager {
       })
     }
   }
+
+  emitDirtyQueriesIds = (): void => {
+    const dirtyQueries = store.getState().queries
+      .dirtyQueriesListToEmitViaWebsocket
+
+    if (dirtyQueries.length > 0) {
+      if (this.websocket) {
+        this.websocket.send({
+          type: `dirtyQueries`,
+          payload: { dirtyQueries },
+        })
+      }
+
+      store.dispatch(clearDirtyQueriesListToEmitViaWebsocket())
```

Comment:
Should we have this inside `if (this.websocket)` above?

Reply:
Not sure TBH,
it might make sense to keep until websocket is alive (so move to inside that conditional block) in case users already have their runtime caches saturated , but I also am not exactly sure if users can even see the page until `this.websocket` is truthy - see https://github.com/gatsbyjs/gatsby/blob/23b413741e0ea4079752a21f0c390ae590d5bc26/packages/gatsby/src/utils/start-server.ts#L340-L345 which init websocket before `http` server even starts to listen on port

But this also assumes that above will never change and putting that `clear` action dispatch inside won't cause any problems other than maybe emitting more than needed?

Reply:
Hmm actually - let's move it there, because users can have their browser caches saturated and restart the dev server - in which case we would drop quite a bit of dirty queries ids so user would have weird stale data until they refresh